config/exclude.txt

```diff
@@ -15,3 +15,6 @@ Shandong/IVDC-SD-001/2020
 Beijing/IVDC-BJ-005/2020
 Henan/IVDC-HeN-002/2020
 Shanghai/IVDC-SH-001/2020
+
+# These exist as duplicate strain IDs on GISAID with different `gisaid_epi_isl`s
+Italy/INMI1/2020
```

Comment:
Despite my adding this to the `exclude.txt`, Snakemake still fails with the following message:
```
ERROR: Problem reading in data/sequences.fasta:
Duplicate key 'Italy/INMI1/2020'
```

Reply:
This error is coming from Bio.SeqIO via `augur filter` when it parses the fasta file, before any exclusions or other filtering happens, so using `exclude.txt` won't work:

https://github.com/nextstrain/augur/blob/bd73cd670add721d73659355850540e80aefe306/augur/filter.py#L111-L116

So I guess either the fasta generation needs to suppress the duplicates or use an actually-unique key, like `gisaid_epi_isl`?

Reply:
@trvrb what would you prefer we do here?

Reply:
The next issue with Bio.SeqIO is that it doesn't like spaces in strain name

Reply:
Fasta definition lines have this syntax:

```
>{id-without-spaces} {freeform description after that}
```

Can you give an example of what you mean by "doesn't like spaces in strain name"?

Reply:
An example strain name from GISAID with spaces is `BetaCoV/Hong Kong/VM20001464/2020`.

The resulting fasta we generate has the related definition line:
`>Hong Kong/VM20001464/2020`

Reply:
Nod, so SeqIO will treat that as a sequence with id `Hong`, as you've discovered. I believe for flu what's been done for Nextstrain builds is to strip spaces → `HongKong`. Other option is to replace spaces with underscore.

(In the future, we could add options to Augur to have it use the full FASTA defline instead of just the id, for cases like this where it is appropriate.)

Reply:
Ok, I stripped spaces for now in `ncov-ingest` to be consistent with what's been done for Nextstrain builds.

Reply:
Stripping whitespace from strain names is good.